### PR TITLE
Introduces World struct

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### Description of the Change

Introduces World struct to improve top level ISI design (e.g. `Register<World, Domain>`). Previously to register the domain for example we had to do the following `Register<Peer, Domain>` and supply `PeerId` in there, but this peer id would be useless and will only confuse users. Instead now this instructions have `World` as their target.

Also this PR removes `Peer` from `WorldStateView` as the peer id is not needed there.

### CODE REVIEW

**Review by MizukiSonoko — COMMENTED** [review]
Nice! Is it not necessary to update the whitepaper? about `Data Model` ?

**Comment by eadventurous** [comment]
> Nice! Is it not necessary to update the whitepaper? about `Data Model` ?

Yes, thank you. I will correct this part in the whitepaper.

**Review by MizukiSonoko — APPROVED** [review]
Thanks!

**Review by humb1t — APPROVED** [review]

File: iroha/benches/validation.rs [hunk 1]
```diff
@@ -141,10 +122,10 @@ fn validate_transaction(criterion: &mut Criterion) {
     .expect("Failed to accept transaction.");
     let mut success_count = 0;
     let mut failures_count = 0;
-    let mut world_state_view = WorldStateView::new(Peer::new(PeerId::new(
-        "127.0.0.1:8080",
-        &key_pair.public_key,
-    )));
+    let mut world_state_view = WorldStateView::new(
+        Peer::new(PeerId::new("127.0.0.1:8080", &key_pair.public_key)),
```
Comment (takemiyamakoto) [inline iroha/benches/validation.rs:126]: I don't understand why peer is here.

File: iroha/src/isi.rs [hunk 2]
```diff
@@ -78,12 +78,11 @@ impl Execute for RegisterBox {
                 }
                 _ => Err("Unsupported instruction.".to_string()),
             },
-            IdBox::PeerId(peer_id) => match self.object {
-                IdentifiableBox::Domain(domain) => Register::<Peer, Domain>::new(*domain, peer_id)
+            IdBox::WorldId => match self.object {
+                IdentifiableBox::Domain(domain) => Register::<World, Domain>::new(*domain, WorldId)
+                    .execute(authority, world_state_view),
+                IdentifiableBox::Peer(peer) => Register::<World, Peer>::new(*peer, WorldId)
```
Comment (takemiyamakoto) [inline iroha/src/isi.rs:84]: I'm not sure why Peer is here either. Generally a peer can be a validating or an observing peer. But the validation or observation is done with respect to an Account that is associated with a Peer. So all permissions are on the Peer's Account and not necessarily the peer itself.

File: iroha/benches/validation.rs (hunk 1 above)
Comment (eadventurous) [reply iroha/benches/validation.rs:126]: As we discussed in the chat, I removed Peer now from WSV, as it is no longer needed there.

File: iroha/src/isi.rs (hunk 2 above)
Comment (eadventurous) [reply iroha/src/isi.rs:84]: Yes, you are right. Though the exact staking mechanism is not yet implemented and Peer keys are not yet associated with account. I think this needs to be done in a separate PR as a separate task.

Right now we simply `Register` a new peer and as soon as the block with this ISI is committed, the peer starts participating in consensus.

**Pull request merged**